src/ExecutionContext.sol

```diff
@@ -40,6 +40,12 @@ library ExecutionContext {
     // context value that should be written to the executionContext storage pointer:
     // executionContext = executionContext.setChecksInProgress();
 
+    function initialize(EC context) internal pure returns (EC result) {
+        // prepopulate the execution context storage slot to optimize gas consumption
+        // (it should never be cleared again thanks to the stamp)
+        result = EC.wrap(EC.unwrap(context) | (STAMP_DUMMY_VALUE << STAMP_OFFSET));
```

Comment:
Why did you make this a method on an existing EC? The previous way seems more intuitive (ie, you're not unwrapping an uninitialised context).

Reply:
This approach makes it so that it doesn't matter when you call the `initialize` function, although it our case it didn't really matter. also, I wanted to make it consistent with `Set.initialize` which theoretically also operates on an uninitialized `SetStorage` data structure